robot/web/web.go

```diff
@@ -845,8 +930,16 @@ func (svc *webService) initSubtypeServices(ctx context.Context) error {
 			subtypeSvc = newSvc
 			svc.services[s] = newSvc
 		}
-		if err := rs.RegisterRPCService(ctx, svc.rpcServer, subtypeSvc); err != nil {
-			return err
+
+		if rs.RegisterRPCService != nil && svc.rpcServer != nil && !mod {
```

Comment:
fold into a larger block of `rs.RegisterRPCService != nil && svc.rpcServer != nil` with a nested mod conditional